gearcatalog/frontend/src/components/gear/EditGearForm.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { EffectCallback, useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { editGear, selectGear } from '../../actions/gear-list';
+import { Link, useHistory } from 'react-router-dom';
+import { useInput } from '../../hooks/useInput';
+import Input from '../common/Input';
+import { Gear } from './types';
+import { RootReducerState } from '../../reducers/reducer.types';
+
+type EditGearProps = {
+  editGear: (gear: Gear, token: string) => void,
+  selectGear: (id: number, token: string) => EffectCallback,
+  selectedGear: Gear,
+  token: string,
+  match: {
+    params: {
+      id: number,
+    },
+  },
+};
+
+const EditGearForm = ({ editGear, selectGear, selectedGear, token,
+match: { params } }: EditGearProps) => {
+  const history = useHistory();
+  const [gearUpdated, setGearUpdated] = useState(false);
+  const [name, setName] = useState(selectedGear.name);
+  const [desc, setDesc] = useState(selectedGear.desc);
+  const [brand, setBrand] = useState(selectedGear.brand);
+  const [weight_grams, setWeight] = useState(selectedGear.weight_grams);
+  const [length_mm, setLength] = useState(selectedGear.length_mm);
+  const [width_mm, setWidth] = useState(selectedGear.width_mm);
+  const [depth_mm, setDepth] = useState(selectedGear.depth_mm);
+  const { value:locking, bind:bindLocking, reset:resetLocking } = 
+  useInput(selectedGear.locking);
```

Comment:
Is this the formatting generated by your linter? It seems a bit odd to have wrapped this without indenting it.

Reply:
No, I was just unsure if I needed to indent lines like that. I will indent it.